### PR TITLE
[GraphQL/Data] IOwner.defaultSuinsName

### PR DESCRIPTION
**Pull request by amnn**
## Description

Rename `IOwner.defaultNameServiceName` to `IOwner.defaultSuinsName` as per feedack from our recent dogfooding session, port the implementation to the new data framework, and apply doc comments consistently.

## Test Plan

Tested manually with:

```
{
  resolveSuinsAddress(domain: "example.sui") {
    address
    defaultNameServiceName
  }

  object(address: "0x55936857d1ca4d8a107b35e4d513bd0932da329389c976e8db6d52c58f16aefb") {
    asMoveObject {
      contents {
        json
      }

      asSuinsRegistration {
        domain
      }
    }
  }
}
```

image

## Stack

- #15749 
- #15760

### CODE REVIEW

**Review by wlmyng — APPROVED** [review]

**Comment by wlmyng** [comment]
the pr description shows manual testing for resolveSuinsAddress instead of defaultSuinsName btw

**Comment by amnn** [comment]
It's both!

File: crates/sui-graphql-rpc/src/types/address.rs [hunk 1]
```diff
@@ -164,11 +164,16 @@ impl Address {
             .extend()
     }
 
-    pub async fn default_name_service_name(&self, ctx: &Context<'_>) -> Result<Option<String>> {
-        ctx.data_unchecked::<PgManager>()
-            .default_name_service_name(ctx.data_unchecked::<NameServiceConfig>(), self.address)
-            .await
-            .extend()
+    /// The domain that a user address has explicitly configured as their default domain
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/types/address.rs:167]:
```suggestion
    /// The domain that a user address has explicitly configured as their default domain.
```

File: crates/sui-graphql-rpc/src/types/object.rs [hunk 2]
```diff
@@ -378,11 +378,15 @@ impl Object {
     }
 
     /// The domain that a user address has explicitly configured as their default domain
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/types/object.rs:380]:
```suggestion
    /// The domain that a user address has explicitly configured as their default domain.
```

File: crates/sui-graphql-rpc/src/types/owner.rs [hunk 3]
```diff
@@ -213,11 +214,16 @@ impl Owner {
             .extend()
     }
 
-    pub async fn default_name_service_name(&self, ctx: &Context<'_>) -> Result<Option<String>> {
-        ctx.data_unchecked::<PgManager>()
-            .default_name_service_name(ctx.data_unchecked::<NameServiceConfig>(), self.address)
-            .await
-            .extend()
+    /// The domain that a user address has explicitly configured as their default domain
```
Comment (stefan-mysten) [inline crates/sui-graphql-rpc/src/types/owner.rs:217]:
```suggestion
    /// The domain that a user address has explicitly configured as their default domain.
```

**Review by stefan-mysten — APPROVED** [review]

**Pull request merged**